### PR TITLE
Review changes - Avoiding catching generic exceptions

### PR DESCRIPTION
**Pull request by sambaheerathan**

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIScopeTestCase.java [hunk 1]
```diff
@@ -142,76 +194,117 @@ public void testSetScopeToResourceTestCase() throws Exception {
...
         // Accessing POST method
-        endPointURL = new URL(gatewayUrls.getWebAppURLNhttp() + "testScopeAPI/1.0.0/PRODUCT/35");
-        response = HttpRequestUtil.doPost(endPointURL, "<resource><PRICE>8.5</PRICE></resource>", requestHeaders);
-        assertEquals(response.getResponseCode(), Response.Status.OK.getStatusCode(),
-                "Admin user cannot access the POST Method");
+        try {
+            endPointURL = new URL(gatewayUrls.getWebAppURLNhttp() + "testScopeAPI/1.0.0/PRODUCT/35");
+            response = HttpRequestUtil.doPost(endPointURL, "<resource><PRICE>8.5</PRICE></resource>", requestHeaders);
+        } catch (MalformedURLException e) {
+            handleException("Exception during GET Operation ", e);
```
Comment (prabathabey) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIScopeTestCase.java]: Be more descriptive when adding error messages.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIScopeTestCase.java [hunk 2]
```diff
@@ -220,10 +313,21 @@ public void destroy() throws Exception {
...
 
         if (userManagementClient != null) {
-            userManagementClient.deleteUser(USER_JOHN);
-            userManagementClient.deleteRole(SUBSCRIBER_ROLE);
+
+            try {
+                userManagementClient.deleteUser(USER_JOHN);
+                userManagementClient.deleteRole(SUBSCRIBER_ROLE);
+            } catch (RemoteException e) {
+                handleException("Exception occured while removing the artifacts ",e);
```
Comment (prabathabey) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIScopeTestCase.java]: Use proper formatting. For example, proper white spacing need to be added between "," and "e".

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIScopeTestCase.java [hunk 3]
```diff
@@ -220,10 +313,21 @@ public void destroy() throws Exception {
...
+            } catch (UserAdminUserAdminException e) {
+                handleException("xception occured while removing the artifacts ",e);
```
Comment (prabathabey) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIScopeTestCase.java]: The same applies here as well.

Comment (sambaheerathan) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIScopeTestCase.java]: Fixed in https://github.com/sambaheerathan/product-apim/commit/40e4b16561d21abf4834ae5ac2b90a1bfc22675e

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIScopeTestCase.java (hunk 2 above)
Comment (sambaheerathan) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIScopeTestCase.java]: fixed in https://github.com/sambaheerathan/product-apim/commit/40e4b16561d21abf4834ae5ac2b90a1bfc22675e

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIScopeTestCase.java (hunk 1 above)
Comment (sambaheerathan) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIScopeTestCase.java]: Fixed in https://github.com/sambaheerathan/product-apim/commit/40e4b16561d21abf4834ae5ac2b90a1bfc22675e

**Comment by bhathiya** [comment]
Can't merge the PR as the master is changed a bit. Can you please update the PR accordingly?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIScopeTestCase.java [hunk 4]
```diff
@@ -239,20 +314,21 @@ public void destroy() throws Exception {
...
-            userManagementClient.deleteUser(USER_JOHN);
-            userManagementClient.deleteRole(SUBSCRIBER_ROLE);
+
+            try {
+                userManagementClient.deleteUser(USER_JOHN);
+                userManagementClient.deleteRole(SUBSCRIBER_ROLE);
+            } catch (RemoteException e) {
+                handleException("Exception occured while removing the artifacts ", e);
+            } catch (UserAdminUserAdminException e) {
+                handleException("xception occured while removing the artifacts ", e);
```
Comment (imesh) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIScopeTestCase.java]: Typo: xception occured

Comment (sambaheerathan) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIScopeTestCase.java]: Currently our code is in frozen state due to release i will fix this once its released

**Pull request closed without merging**